### ISSUE TITLE
Global unit not running on all machines

### ISSUE THREAD

**Issue by aburnett** [coreos/fleet#1310]
I have a global unit deployed to my cluster, but for some reason it will not deploy to one node in the cluster. Where do I start troubleshooting this?

**Unit**

    [Unit]
    Description=Analyzes resource usage and performance characteristics of running containers.
    After=docker.service
    Requires=docker.service

    [Service]
    Restart=always
    ExecStartPre=-/usr/bin/docker kill %n
    ExecStartPre=-/usr/bin/docker rm -f %n 
    ExecStartPre=/usr/bin/docker pull google/cadvisor:latest
    ExecStart=/usr/bin/docker run --volume=/var/run:/var/run:rw --volume=/sys/fs/cgroup/:/sys/fs/cgroup:ro --volume=/var/lib/docker/:/var/lib/docker:ro --publish=8080:8080 --name %n google/cadvisor:latest
    ExecStop=-/usr/bin/docker stop %n

    [X-Fleet]
    Global=true

**Cluster**

    $ fleetctl list-machines
    MACHINE		IP		METADATA
    869b5e94...	10.12.12.223	-
    943b0e30...	10.12.12.221	-
    e081671e...	10.12.12.222	-

**After starting** - units are only loaded on 222, and 223 but not 221

    $ fleetctl list-units
    UNIT					MACHINE				ACTIVE		SUB
    cadvisor.service			869b5e94.../10.12.12.223	active		running
    cadvisor.service			e081671e.../10.12.12.222	active		running
    ...

**Comment by mischief** [comment]
can you check `fleetctl ssh 943b0e30 systemctl status cadvisor.service`?

**Comment by aburnett** [comment]
The unit isn't loaded there.

    $ systemctl status cadvisor.service
    ● cadvisor.service
       Loaded: not-found (Reason: No such file or directory)
       Active: inactive (dead)

Also strange, if I try pinning a service to `943b0e30` using `MachineID` then fleetctl just hangs trying to load the unit. It seems like fleet doesn't consider `943b0e30` to be part of the cluster?

**Comment by aburnett** [comment]
So I found the problem, but I don't know why it happened. The root cause was a bad unit file in `/run/fleet/units` on `943b0e30`. Deleting the file un-wedged fleet on that host and all started going well.

What I don't know is why that unit file was there in the first place and why a single unit blocked all other units from running. I had used `fleetctl destroy` on it before.

    Jul 16 13:34:35 vtse-coreos01 fleetd[31875]: ERROR reconcile.go:79: Unable to determine agent's current state: failed fetching unit states from UnitManager: Unit name registrator@.service is not valid
